web/impact/impact/v1/views/cancel_office_hour_session_view.py

```diff
@@ -0,0 +1,126 @@
+from pytz import timezone
+
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from django.template import loader
+
+from rest_framework.exceptions import PermissionDenied
+from rest_framework.generics import get_object_or_404
+from rest_framework.response import Response
+
+from accelerator.models import MentorProgramOfficeHour
+from impact.minimal_email_handler import MinimalEmailHandler
+from impact.permissions.v1_api_permissions import OfficeHourPermission
+from impact.v1.views.impact_view import ImpactView
+
+User = get_user_model()
+
+DEFAULT_TIMEZONE = 'UTC'
+SUBJECT = '[Office Hours] Canceled: {date}, {start_time}'
+
+STAFF_NOTIFICATION = 'Canceled office hours on behalf of ' \
+                     '{mentor_name} at {start_time} - {end_time} on {date}'
+MENTOR_NOTIFICATION = 'Canceled office hours at {start_time} - ' \
+                      '{end_time} on {date}'
+PERMISSION_DENIED = 'Office hour session could not be canceled. ' \
+                    'You do not have permission to cancel that session'
+
+
+def get_office_hours_list_url(family_slug, program_slug):
+    site_url = 'accelerate.masschallenge.org'
+    return f'https://{site_url}/officehours/list/{family_slug}/{program_slug}'
+
+
+def get_office_hour_shared_context(office_hour, message=None):
+    family_slug = office_hour.program.program_family.url_slug
+    program_slug = office_hour.program.url_slug
+    tz = timezone(office_hour.location.timezone or DEFAULT_TIMEZONE)
+    date = office_hour.start_date_time.astimezone(tz).strftime('%A, %d %B, %Y')
+    start_time = office_hour.start_date_time.astimezone(tz).strftime('%I:%M%p')
+    end_time = office_hour.end_date_time.astimezone(tz).strftime('%I:%M%p')
+    return {
+        'date': date,
+        'start_time': start_time,
+        'end_time': end_time,
+        'location': office_hour.location.name if office_hour.location else '',
+        'dashboard_url': get_office_hours_list_url(family_slug, program_slug),
+        'mentor_name': office_hour.mentor.get_profile().full_name(),
+        'phone': office_hour.program.program_family.phone_number,
+        'message': message,
+    }
+
+
+def get_ui_notification(context, staff=False):
+    if staff:
```

Comment:
Not a blocker, but dicts are preferable over ifs for data lookup, since they are extensible.
For example, if we decided to notify finalists of cancelled sessions, this would need to be changed, but

```
def get_ui_notification(context, staff=False):
    notification = UI_NOTIFICATIONS[participant_role]
    return notification.format(**context)
```

would not.